src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2026,7 +2027,67 @@ protected override void HandleEvaluateRequestAsync(IRequestResponder<EvaluateArg
                 Type = variable.Type,
                 VariablesReference = variable.VariablesReference
             });
+        }
+
+
+        protected override void HandleReadMemoryRequestAsync(IRequestResponder<ReadMemoryArguments, ReadMemoryResponse> responder)
+        {
+            int hr;
+            ReadMemoryArguments rma = responder.Arguments;
+            ErrorBuilder eb = new ErrorBuilder(() => AD7Resources.Error_Scenario_ReadMemory);
+            try
+            {
+                if (string.IsNullOrEmpty(rma.MemoryReference))
+                {
+                    throw new ArgumentNullException("ReadMemoryArguments.MemoryReference is null or empty.");
+                }
 
+                ulong address;
+                if (rma.MemoryReference.ToLowerInvariant().StartsWith("0x"))
+                {
+                    address = Convert.ToUInt64(rma.MemoryReference.Substring(2), 16);
+                }
+                else
+                {
+                    address = Convert.ToUInt64(rma.MemoryReference, 10);
+                }
+
+                hr = ((IDebugMemoryBytesDAP)m_engine).CreateMemoryContext(address, out IDebugMemoryContext2 ppMemory);
+                eb.CheckHR(hr);
+
+                if (rma.Offset.HasValue && rma.Offset.Value != 0)
+                {
+                    ulong offset = Convert.ToUInt64(Math.Abs(rma.Offset.Value));
```

Comment:
>Convert.ToUInt64(Math.Abs(rma.Offset.Value)); [](start = 35, length = 45)

Any reason to not just cast? (ex: `.Subtract((ulong)(-rma.Offset.Value)`)

Reply:
I was trying to reuse offset for both the additon and subtract case.

Reply:
I would find the casts simpler, but up to you...